server/base/src/main/java/org/apache/accumulo/server/conf/codec/VersionedProperties.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.server.conf.codec;
+
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.util.Collection;
+import java.util.Map;
+
+public interface VersionedProperties {
```

Comment:
I don't think this needs to be an interface with a separate implementation. These two can be combined and simplified to just a simple POJO container of `{ version, propMap }`.

I'm thinking something like:

```java
public class VersionedProperties {
  private final long version;
  private final Map<String,String> properties;

  public VersionedProperties(long version, Map<String,String> properties) {
    this.version = version;
    this.properties = Map.copyOf(requireNonNull(properties));
  }
  
  public long getVersion() {
    return version;
  }
  
  public Map<String,String> getProperties() {
    return properties;
  }
}
```